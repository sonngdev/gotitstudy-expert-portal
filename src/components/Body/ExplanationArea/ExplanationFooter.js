import React from 'react';
import { EndSessionButton } from './EndSessionButton';
import { FlagSessionButton } from './FlagSessionButton';
import { PreviewButton } from './PreviewButton';
import './styles/ExplanationFooter.css';

export function ExplanationFooter() {
  return (
    <div className="explanation-footer u-flex u-justifyContentBetween u-paddingRightSmall">
      <div>
        <EndSessionButton className="u-marginRightSmall" />
        <FlagSessionButton />
      </div>

      <PreviewButton />
    </div>
  );
}
