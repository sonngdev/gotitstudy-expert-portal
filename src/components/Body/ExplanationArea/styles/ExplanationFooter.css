.explanation-footer {
  padding-top: 10px;
}
