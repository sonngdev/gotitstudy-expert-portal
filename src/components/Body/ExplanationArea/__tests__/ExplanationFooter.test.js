import React from 'react';
import { shallow } from 'enzyme';
import { ExplanationFooter } from '../ExplanationFooter';

describe('<ExplanationFooter />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ExplanationFooter />);
  });

  it('renders', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('contains a button to end session', () => {
    expect(wrapper.find('EndSessionButton').exists()).toBe(true);
  });

  it('contains a button to flag session', () => {
    expect(wrapper.find('FlagSessionButton').exists()).toBe(true);
  });

  it('contains a button to preview explanation', () => {
    expect(wrapper.find('PreviewButton').exists()).toBe(true);
  });
});
