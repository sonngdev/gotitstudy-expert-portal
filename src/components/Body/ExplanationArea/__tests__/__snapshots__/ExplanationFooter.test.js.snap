// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExplanationFooter /> renders 1`] = `
"<div className=\\"explanation-footer u-flex u-justifyContentBetween u-paddingRightSmall\\">
  <div>
    <EndSessionButton className=\\"u-marginRightSmall\\" />
    <FlagSessionButton />
  </div>
  <PreviewButton />
</div>"
`;
